### PR TITLE
CloudWatch: Add generic filter component to variable editor

### LINKED ISSUE #47597: CloudWatch: add generic filter builder for Variable Editor
Currently, users enter a text representation of json for the ec2 Instance Attribute and Resource ARN filters, and a builder should be added for them. This shouldn't call getDimensionValues, so it should be a new generic filter builder instead of dimensions

### PR DESCRIPTION
**Pull request by iwysiu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #47597

**Special
 notes for your reviewer**:
The new filter object takes a comma-separated list as the value

old:
Screen Shot 2022-04-20 at 2 20 49 PM

new:
Screen Shot 2022-04-20 at 2 14 07 PM

### CODE REVIEW

**Review by andresmgot — COMMENTED** [review]
Thanks @iwysiu! Good job here. I have a bunch of comments but I am probably missing some context.

As a general comment, when submitting a frontend PR, please include screenshots / gifs / videos of the changes included. That helps with the review and how is the feature meant to work.

Two things I noticed when testing this:

 - When selecting an Attribute Name it immediately fails.

Screenshot from 2022-04-20 11-53-41

 - Also, when writing something in the value but without hitting Tab or Enter, the value disappears (also the key input size keeps growing until a limit, is that something we can avoid?):

Peek 2022-04-20 11-56

Two options for this (but I don't know if the experimental package supports it): Either save the value on blur or show it as an option so it's selectable and clear like:

Screenshot from 2022-04-20 11-55-03

**Review by iwysiu — COMMENTED** [review]
After talking to @andresmgot , we decided to separate creating and using the mulitselect text box into another issue. Now, this pr just takes a comma separated string for multifilter values
We also found that the error he was getting was an authentication issue, and I'm filing another issue to improve the errors sent from the backend.

File: public/app/plugins/datasource/cloudwatch/components/VariableQueryEditor/VariableQueryEditor.tsx [hunk 1]
```diff
@@ -163,14 +164,20 @@ export const VariableQueryEditor = ({ query, datasource, onChange }: Props) => {
             placeholder="attribute name"
             onBlur={(value: string) => onQueryChange({ ...parsedQuery, attributeName: value })}
             label="Attribute Name"
+            tooltip='Attribute or tag to query on. Tags should be formatted "Tags.<name>".'
```
Comment (andresmgot) [inline public/app/plugins/datasource/cloudwatch/components/VariableQueryEditor/VariableQueryEditor.tsx:167]: what about adding a link to the documentation so people know what they can use here?

File: public/app/plugins/datasource/cloudwatch/components/VariableQueryEditor/MultiFilterItem.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  const styles = getOperatorStyles(theme);
+
+  return (
+    <div data-testid="cloudwatch-multifilter-item">
+      <InputGroup>
+        <Input
+          data-testid="cloudwatch-multifilter-item-key"
+          aria-label="Filter key"
+          value={localKey}
+          placeholder="key"
```
Comment (andresmgot) [inline public/app/plugins/datasource/cloudwatch/components/VariableQueryEditor/MultiFilterItem.tsx:27]: can you make the placeholder an optional property? that way the parent component can use more meaningful examples (like an example attribute for the ec2 instance or a tag for the arn resource)

**Review by andresmgot — COMMENTED** [review]
Thanks for the changes! I have some more minor UX suggestions.

Apart from that, I found a bug, when you try to edit the "key" of the MultiFilter, it removes it:

Peek 2022-04-21 10-08

**Review by iwysiu — COMMENTED** [review]
Fixed the bug on key changes by not clearing the value field. I assume that if you want to just change the key you wanted the same value.
Now that 8.5.0's been cut, I added a migration between the released VariableQuery and this one.

**Review by andresmgot — APPROVED** [review]
LGTM, nice job here :)

I am not sure if you have done it already but is there any issue for properly rendering the error returned here? (so it's not needed to check the browser console to get the real error message)

**Comment by iwysiu** [comment]
I realized the template variables weren't working properly, so I fixed them. Unfortunately, due to how the code is structured it's hard to test in variable, but I tested the variable replacement function.
This error doesn't render properly either 😞  I think all of the backend errors are getting turned into 400 errors.

**Comment by andresmgot** [comment]
> This error doesn't render properly either  I think all of the backend errors are getting turned into 400 errors.

Yes, this is the error I meant. We can open a follow-up issue to fix that.

**Review by asimpson — APPROVED** [review]
LGTM once we get the test failures ironed out. Nice work!! :trophy:

**Pull request merged**